src/TypeScript/NodeJSLocator/NodeLocator.cs

```diff
@@ -19,57 +19,53 @@
  */
 
 using System;
-using System.Collections.Generic;
 using System.ComponentModel.Composition;
 using System.Diagnostics;
 using System.IO.Abstractions;
 using SonarLint.VisualStudio.Integration;
 
 namespace SonarLint.VisualStudio.TypeScript.NodeJSLocator
 {
+    internal interface INodeLocator
+    {
+        /// <summary>
+        /// Returns the absolute file path of a compatible `node.exe`, or null if no compatible version was found.
+        /// </summary>
+        string Locate();
+    }
+
     [Export(typeof(INodeLocator))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class NodeLocatorAggregator : INodeLocator
+    internal class NodeLocator : INodeLocator
     {
+        private readonly INodeLocationsProvider nodeLocationsProvider;
         private readonly ILogger logger;
         private readonly IFileSystem fileSystem;
         private readonly Func<string, Version> getNodeExeVersion;
-        private readonly IEnumerable<INodeLocator> nodeLocators;
 
         [ImportingConstructor]
-        public NodeLocatorAggregator(INodeLocatorsProvider nodeLocatorsProvider, ILogger logger)
-            : this(nodeLocatorsProvider, logger, new FileSystem(),  GetNodeVersion)
+        public NodeLocator(INodeLocationsProvider nodeLocationsProvider, ILogger logger)
+            : this(nodeLocationsProvider, logger, new FileSystem(), GetNodeVersion)
         {
         }
 
-        internal NodeLocatorAggregator(INodeLocatorsProvider nodeLocatorsProvider, 
-            ILogger logger, 
+        internal NodeLocator(INodeLocationsProvider nodeLocationsProvider,
+            ILogger logger,
             IFileSystem fileSystem,
             Func<string, Version> getNodeExeVersion)
         {
+            this.nodeLocationsProvider = nodeLocationsProvider;
             this.logger = logger;
             this.fileSystem = fileSystem;
             this.getNodeExeVersion = getNodeExeVersion;
-            nodeLocators = nodeLocatorsProvider.Get();
         }
 
-        /// <summary>
-        /// Returns the absolute file path of a compatible `node.exe`, or null if no compatible version was found.
-        /// </summary>
         public string Locate()
         {
-            foreach (var nodeLocator in nodeLocators)
+            foreach (var nodeExePath in nodeLocationsProvider.Get())
             {
-                var nodeExePath = nodeLocator.Locate();
-
-                if (string.IsNullOrEmpty(nodeExePath))
-                {
-                    continue;
-                }
-
-                if (!fileSystem.File.Exists(nodeExePath))
+                if (string.IsNullOrEmpty(nodeExePath) || !fileSystem.File.Exists(nodeExePath))
                 {
-                    logger.WriteLine(Resources.ERR_FileNotFound, nodeExePath);
```

Comment:
Since I'm now returning all candidate file paths, including the numerous folders in the PATH var, I think it will be too noisy to log all the "not found" attempts.